packages/cadl-ranch-specs/http/models/inheritance-complex/main.cadl

```diff
@@ -0,0 +1,114 @@
+import "@azure-tools/cadl-azure-core";
+import "@azure-tools/cadl-ranch-expect";
+
+using Cadl.Rest;
+using Cadl.Http;
+using Azure.Core.Foundations;
+
+@serviceTitle("ComplexInheritanceClient")
+@serviceVersion("1.0.0")
+@server("http://localhost:3000", "Testserver endpoint")
+@doc("Illustrates complex polymorphic model.")
+@route("/inheritance-complex")
+namespace Model.Inheritance;
+
+@doc("Base model for non-discriminator three levels inheritance.")
+model BaseModel {
+  optionalString?: string;
+}
+
+@doc("The second level model A in the non-discriminator three levels inheritance.")
+model DerivedFromBaseModel extends BaseModel {
+  optionalInt?: int32;
+}
+
+@doc("The third level model A in the non-discriminator three levels inheritance.")
+model BottomModel extends DerivedFromBaseModel {
+  optionalBool?: boolean;
+}
+
+@doc("Base model for three levels inheritance with discriminator.")
+@discriminator("discriminator1")
+model BaseModelWithDiscriminator {
+  optionalInt?: int32;
+}
+
+@doc("The second level model in three levels inheritance with discriminator and defined a new discriminator.")
+@discriminator("discriminator2")
+model SubModelWithDiscriminator extends BaseModelWithDiscriminator {
+  discriminator1: "SubModel";
+}
+
+@doc("The second level model in three levels inheritance with discriminator.")
+model DerivedFromBaseModelWithDiscriminator extends BaseModelWithDiscriminator {
```

Comment:
I understand that you're following the naming schema of the existing `inheritance` models, but I've been meaning to change that one as well, since it's not clear what the models are. @timotheeguerin I honestly think in situations like these, it's more clear to name these models things like `Animal`, `Pet`, `Dog`, `Puppy` for example, since you can easily visualize the relationship between these models. The scenario names shouldn't include references to dogs etc, but I think it makes sense for the models to have a metaphorical relationship. What are your thoughts?

Reply:
yeah the models can have those names, just the scenarios names should be clear about whwat they do. The implementation details of a scenario(like model names) should be tailored to make the life easy for generator looking to implement the scenario

Reply:
renamed model&property names to make the model more easy to understand